manim/utils/config_utils.py

```diff
@@ -130,11 +130,23 @@ def _parse_file_writer_config(config_parser, args):
         ]
     }
 
+
+
     # For internal use (no CLI flag)
     fw_config["skip_animations"] = any(
         [fw_config["save_last_frame"], fw_config["from_animation_number"]]
     )
 
+    # the the custom_folders flag overrides the default folder structure with
+    # the custom folders defined in default.cfg
+    if args.custom_folders:
+        from pathlib import Path
+        root_folder = str(config_parser['custom_folders'].get("custom_root"))
+        for opt in ['video_dir', 'tex_dir', 'text_dir', 'output_file']:
+            sub_folder = config_parser['custom_folders'].get(opt)
+            path = Path.home() / root_folder / sub_folder
```

Comment:
Issue: I am (weakly) against having all manim videos go to the user's home. At the very least, we should document that explicitly somewhere. What do others think?

Reply:
That workflow makes a lot of sense. HOWEVER, (and this is a BIG one): please do not quote your own workflow as the reason for making library-wide changes to manim. When changing manim (and **especially** when changing the `default.cfg` file), you are changing how manim works for EVERYONE. There are dozens (perhaps hundreds? maybe thousands?) of people who use manim, and your personal workflow may not be adequate for all of their needs.

The fact that the home folder is convenient **for you** is NOT a good reason to make manim use the home folder for **every user**.

Reply:
That is something to keep in mind.
What would you recommend instead as a starting point for the folder structure?
Further: I am aware that other users will have other workflows, I did not want to sound like I want everyone to use it the way I use it. My idea was to implement it in a general way (which might have the tendency to be convenient for me), but with the capability to be easily changed for others. The reason I choose the home folder was, that it is quit platform-independent (I guess ubuntu/mac/windows all have a home folder where pathlib can point to).

Reply:
I have nothing against your workflow! Like I said, it makes sense to me. What I'm trying to point out is that you are submitting this PR, which will end up being used by every user (if it gets merged). Therefore, you are responsible for convincing the dev team and the rest of the community, that your PR makes sense for every user. You tried to do that by quoting your personal workflow, which is not an acceptable answer.

So, you have to support the choices in this PR from a point of view of general convenience to everyone, not by using your personal preferences as your main reasons :) .

I'm still weakly against having the home be the default, and I don't hear any arguments in favor.

Reply:
So far manim has just put the video in the directory where it's run. Leaving it there will preserve consistency and eliminate any potential surprises where people don't know where the video is written.

Reply:
I'm against having the home be the default. 
For one main reason, that @eulertour pointed out, which is the 

> potential surprises where people don't know where the video is written.

I think we should keep the default directory as it is currently, ie the directory where manim is run.